### DISCUSSION TITLE
[Enhancement] 能否在`el-config-provider`提供全局的插槽和加载占位

### DISCUSSION THREAD

**Discussion by zhixiaotong** [element-plus/element-plus#15548] [Feature Request]
Feature type: **`Enhancement`**

## Details

- Enhanced component name: `N/A`

### 背景
我从`element-ui`到现在，深度使用`ep`。目前发现一些不好用的地方，放在这里，`ep`团队的各位看看我这个想法是否合理？

### 问题
由于表格的加载和其他加载不同。在项目或产品的角度，统一视觉风格是重要的。目前需要统一`loading`和`empty`的效果，我就需要在产品中几百个`vue`文件中，`Table`、`Select`、`SelectV2`、`Tree`、`TreeSelect`、`autocomplete`等等常用组件下（被调用四五百次，甚至更多~）挨个添加类似下面的代码。冗余不说，不好用是真的。

```ts
// ...
<template #loading>
  <x-loading />
</template>

// ...
<template #empty>
  <x-empty/>
</template>
```

### 一个不成熟的建议
能否在`el-config-provider`添加全局的`loading`和`empty`的处理，如果部分组件内部有重写，则以重写的为准。此法可大幅度降低在实际应用中编写冗余代码的情况。

### Additional comments

_(empty)_

**Comment by zhixiaotong** [comment]
@kooriookami 麻烦你先看一下这个需求，还可以艾特其他官方的人看一下，谢谢~

**Reply by kooriookami** [reply]
OK~

**Comment by kooriookami** [comment]
默认slots和默认props是类似的需求
https://github.com/element-plus/element-plus/issues/15189#issuecomment-1893514007

**Reply by zhixiaotong** [reply]
看到了，其他人也有这样类似的需求，可不知道之前为什么会未通过(′⌒‘)。希望团队的人考虑下，使用全局的方式能解决这个痛点，不限于`config-provider`的方式

**Reply by zhixiaotong** [reply]
应用中一般不会直接用`ep`自带的`empty`和`loading`效果，`ep`在使用的时候一般默认样式都会做一些调整，当做覆盖处理的时候，大量组件都会存在这个问题，单个覆盖可以接受，几百个页面，一个表单页面都有好几个`select`，全局添加插槽对开发者不友好，且代码量较多，意义不大，还容易遗漏，造成不好用的局面。
